client/src/SubmitBidForm.js

```diff
@@ -75,13 +75,13 @@ const SubmitBidForm = () => {
 				alert("Bid submitted successfully!");
 				navigate("/dashboard");
 			} catch (error) {
-				const { status, data } = error.response;
+				const { status, data } = error.response || {};
 				if (status === 400) {
 					setBidError("Validation error");
-					setErrors(data.error);
+					setErrors(data?.error || ["An unknown validation error occurred"]);
```

Comment:
I just noticed the api doesn't follow our standard. All validation errors should be returned in the response. You can look at `/signup` for an example of how it should be done

```suggestion
					setErrors(data.errors);
```